scaffold/pattern-extract.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * Pattern Extract Script
+ *
+ * Extracts and outputs a specified block pattern or post content as JSON, identified by name via WP-CLI.
+ * Intended for use within a WordPress environment.
+ */
+
+// Check if an argument is provided, otherwise exit.
+if ( empty( $args ) ) {
+	exit( 1 );
+}
+
+$pattern_name = $args[0];
+
+// Attempt to retrieve the registered block pattern.
+$pattern_registry = WP_Block_Patterns_Registry::get_instance();
+$pattern          = $pattern_registry->get_registered( $pattern_name );
+
+$result = array();
```

Comment:
This is extremely nit-picky but what if `$result` was initially `null`? And we changed the checks down to `is_null` instead? Currently, I thought that `$result` will be populated with multiple entries but that's not the case -- we are using the empty array as a null-ish value which works great but I had to look at the code twice because I thought it's some sort of mistake in assignment on lines 24 and 39.

Reply:
I could see that getting confusing. `empty()` actually acts the same way with an empty array or a null value, but I updated the check to `is_null()` as well to help avoid that kind of confusion.

Updated in f8342d3